packages/mobile/src/account/LinkBankAccountScreen.tsx

```diff
@@ -22,92 +26,167 @@ import VerificationPending from 'src/icons/VerificationPending'
+        <StepOne />
+        <StepTwo />
       </ScrollView>
     </SafeAreaView>
   )
 }
 
-export function StepOne({ kycStatus }: StepOneProps) {
+interface StepOneUIStateParams {
+  kycStatus: KycStatus | undefined
+  finclusiveKycStatus: FinclusiveKycStatus
+  errorFromPersona: boolean
+  successFromPersona: boolean
+  isInPersonaFlow: boolean
+}
+
+enum StepOneUIState {
+  Begin = 'Begin',
+  Failure = 'Failure',
+  Pending = 'Pending',
+  Completed = 'Completed',
+  Spinner = 'Spinner',
+}
+export function stepOneUIState({
+  kycStatus,
+  finclusiveKycStatus,
+  errorFromPersona,
+  successFromPersona,
+  isInPersonaFlow,
+}: StepOneUIStateParams): StepOneUIState {
+  // The persona onSuccess / onError hooks fire slightly after the user is bounced back
+  // to the LinkedBankAccount Screen. We show a spinner at this time to avoid a more
+  // unpleasent screen flash as the kyc states change
+  if (isInPersonaFlow) {
+    return StepOneUIState.Spinner
+  }
+
+  // Show the Completed Screen
+  if (finclusiveKycStatus === FinclusiveKycStatus.Accepted) {
+    return StepOneUIState.Completed
+  }
+
+  // Show the Pending Screen
+  const userCompletedPersona: (KycStatus | undefined)[] = [
+    KycStatus.Completed,
+    KycStatus.Approved,
+    KycStatus.NeedsReview,
+  ]
+  const finclusiveNotCompleted = [FinclusiveKycStatus.Submitted, FinclusiveKycStatus.InReview]
+  if (
+    successFromPersona ||
+    userCompletedPersona.includes(kycStatus) ||
+    finclusiveNotCompleted.includes(finclusiveKycStatus)
+  ) {
+    return StepOneUIState.Pending
+  }
+
+  // Show the Failed Screen
+  const personaFailed: (KycStatus | undefined)[] = [KycStatus.Failed, KycStatus.Declined]
+  if (
+    errorFromPersona ||
+    personaFailed.includes(kycStatus) ||
+    finclusiveKycStatus === FinclusiveKycStatus.Rejected
+  ) {
+    return StepOneUIState.Failure
+  }
+
+  // Show Begin Screen
+  return StepOneUIState.Begin
+}
+
+export function StepOne() {
   const { t } = useTranslation()
-  const [isKycVerifying, setIsKycVerifying] = useState(false)
+  const dispatch = useDispatch()
+  const [isInPersonaFlow, setIsInPersonaFlow] = useState(false)
+  const [errorFromPersona, setErrorFromPersona] = useState(false)
+  const [successFromPersona, setSuccessFromPersona] = useState(false)
+  const kycStatus = useSelector(kycStatusSelector)
+  const finclusiveKycStatus = useSelector(finclusiveKycStatusSelector)
 
+  // If the user's KYC has been accepted by Persona but not by Finclusive, then continuously
+  // poll finclusive for updates until the KYC is accepted
+  const pollFinclusiveKyc = () => {
+    if (kycStatus === KycStatus.Approved && finclusiveKycStatus !== FinclusiveKycStatus.Accepted) {
+      dispatch(fetchFinclusiveKyc())
+    }
+  }
+  useEffect(() => {
+    if (kycStatus === KycStatus.Approved && finclusiveKycStatus !== FinclusiveKycStatus.Accepted) {
+      pollFinclusiveKyc()
```

Comment:
i dont think this line is needed. setInterval() already calls the function.

Reply:
setInterval calls pollFinclusiveKyc after [interval]. So this call is needed so that the function is called right when the user sees the LinkBankAccountsScreen